browser/ui/webui/new_tab_page/top_sites_message_handler.cc

```diff
@@ -356,8 +349,12 @@ void TopSitesMessageHandler::HandleAddNewTopSite(
   if (!GetValidURLStringForTopSite(&url))
     return;
 
-  // similar to `MostVisitedHandler::AddMostVisitedTile`
-  if (most_visited_sites_->IsCustomLinksEnabled()) {
-    most_visited_sites_->AddCustomLink(GURL(url), base::UTF8ToUTF16(title));
```

Comment:
So, this seems to be the cause! Didn't add custom link when it's in MV mode.

Reply:
Yeah, it was meant to convert to `CustomLinks` mode and then add it but it wasn't :smile: I was talking to rebron, and hopefully we're going to have a bigger rethink of how the new tab page grid is going to work, which should be exciting!